external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/JsonUtils.scala

```diff
@@ -106,13 +107,23 @@ private object JsonUtils {
     Serialization.write(result)
   }
 
-  def topicTimestamps(topicTimestamps: Map[String, Long]): String = {
-    val result = new HashMap[String, Long]()
-    val topics = topicTimestamps.keySet.toSeq.sorted  // sort for more determinism
-    topics.foreach { topic =>
-      val timestamp = topicTimestamps(topic)
-      result += topic -> timestamp
+  def partitionTimestamps(topicTimestamps: Map[TopicPartition, Long]): String = {
```

Comment:
At the first glance don't see the difference compared to `partitionOffsets` (apart from the incoming parameter name).

Reply:
That is now same but different meaning so I wouldn't remove the method. Maybe I'll call `partitionOffsets` here for now and fix here if we have to make a difference.